incubator/sparkoperator/templates/spark-operator.yaml

```diff
@@ -0,0 +1,170 @@
+# If the admission webhook is enabled, then the following pre-install steps are performed
+# 1. Create the serviceAccount "sparkoperator"
+# 2. Create the namespace ("sparkoperator" by default)
+# 3. Generate and install the secret in the namespace created in step 2.
+
+# Steps above need to be executed in the order presented because the first two steps are the pre-requisites
+# for step 3 (e.g. In step 3, the token corresponding to the service account created in step 1 is used to communicate
+# with the Kubernetes API server to install the secret bundle. The order are specified using helm.sh/hook-weight
+# annotations.
+
+{{ if .Values.enableWebhook }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.operatorNamespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-10"
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: sparkoperator
+  namespace: {{ .Values.operatorNamespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: sparkoperator
+  namespace: {{ .Values.operatorNamespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "0"
+  labels:
+    app.kubernetes.io/name: sparkoperator
+    app.kubernetes.io/version: {{ .Values.operatorVersion }}
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: sparkoperator
+        app.kubernetes.io/version: {{ .Values.operatorVersion }}
+    spec:
+      serviceAccountName: sparkoperator
+      restartPolicy: OnFailure
+      containers:
+      - name: main
+        image: {{ .Values.operatorImageName }}:{{ .Values.operatorVersion }}
+        imagePullPolicy: Always
+        command: ["/usr/bin/gencerts.sh", "-n", "{{ .Values.operatorNamespace }}", "-p"]
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: sparkoperator-cleanup
```

Comment:
I can see that cleanup job runs without a namespace, but init job runs in a namespace. I think I encounter the same problem with the namespace deletion as with secret creation. Investigating now

Reply:
I put the cleanup job to run in the default namespace because I don't want the job to run in the namespace that it's trying to delete. The `"helm.sh/hook-delete-policy": hook-succeeded` ensures that the cleanup job itself is cleaned up after it's done. But I haven't been to reproduce errors you saw. I wonder if it's something that I have in my cluster but I forgot..

Reply:
I run the cleanup job separately, and I get this
```
HTTP/1.1 403 Forbidden
Content-Type: application/json
X-Content-Type-Options: nosniff
Date: Wed, 22 Aug 2018 16:11:36 GMT
Content-Length: 376

{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "namespaces \"spark-operator\" is forbidden: User \"system:serviceaccount:default:default\" cannot delete namespaces in the namespace \"spark-operator\"",
  "reason": "Forbidden",
  "details": {
    "name": "spark-operator",
    "kind": "namespaces"
  },
  "code": 403
}
```